File: CarTest/car.py
class Car:
    def _init_(self, fuel_efficiency):
       
    
        self.fuel_efficiency = fuel_efficiency
        self.fuel_level = 0
        self.running = False
        self.max_capacity = 50

    def start(self):
        if self.fuel_level > 0:
            self.running = True

    def stop(self):
        self.running = False

    def refuel(self, liters):
        
        self.fuel_level = min(self.fuel_level + liters, self.max_capacity)

    def drive(self, distance):
        
        if not self.running:
            raise Exception("Car must be started before driving")

        fuel_needed = distance / self.fuel_efficiency
        if fuel_needed > self.fuel_level:
            raise Exception("Not enough fuel to drive the distance")

        self.fuel_level -= fuel_needed

   
    def get_fuel_level(self):
        return self.fuel_level

    def is_running(self):
        return self.running

File: CarTest/Car.java
public class Car {
    private final int fuelEfficiency; // km per liter
    private int fuelLevel;
    private boolean running;
    private final int maxCapacity = 50;

    public Car(int fuelEfficiency) {
        this.fuelEfficiency = fuelEfficiency;
        this.fuelLevel = 0;
        this.running = false;
    }

    public void start() {
        if (fuelLevel > 0) {
            running = true;
        }
    }

    public void stop() {
        running = false;
    }

    public void refuel(int liters) {
        fuelLevel = Math.min(fuelLevel + liters, maxCapacity);
    }

    public void drive(int distance) {
        if (!running) {
            throw new IllegalStateException("Car must be started before driving");
        }

        int fuelNeeded = distance / fuelEfficiency;
        if (fuelNeeded > fuelLevel) {
            throw new IllegalStateException("Not enough fuel to drive the distance");
        }

        fuelLevel -= fuelNeeded;
    }

    
    public int getFuelLevel() {
        return fuelLevel;
    }

    public boolean isRunning() {
        return running;
    }
}